### PR TITLE
Add initial version of the script and docs.

### PR DESCRIPTION
**Pull request by artem-zinnatullin**

### CODE REVIEW

**Comment by AntonRutkevich** [comment]
Give me 1 day for review plz, don't merge yet.

File: mainframer.sh [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+rm -rf android-project-remote-build/build/remotely_built_project.tar android-project-remote-build/*/src && \
+tar -xzf project_for_remote_build.tar -C android-project-remote-build && \
+cd android-project-remote-build && \
+$BUILD_COMMAND && \
+GZIP=-$REMOTE_GZIP_LEVEL tar -cz \
+--exclude='kotlin' \
+--exclude='tmp' \
+-f remotely_built_project.tar build/ */build"
+
+# Clean local build dirs.
```
Comment (dmitry-novikov) [inline mainframer.sh]: What about changing order of this and next step? The idea is to reduce time without any build  files for Studio (any other IDE) to a minimum possible.

Comment (ming13) [inline mainframer.sh]: `rm` is cheap, but can be an issue on may files indeed. `parallel` would be nice, but requires a dependency.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+#### Info you'll need to start
+To use this Remote Build script You will need to receive following information before proceeding.
+
+* `BUILD_MACHINE_NAME` — build machine hostname.
+* `BUILD_MACHINE_IP_OR_HOSTNAME` — build machine IP or hostname. Something like `172.42.42.42` or `woooot.remotebuild.mycompany.com`.
+* `BUILD_MACHINE_USERNAME` — your build machine user. Something like `john_doe`.
+
+# Authentication
+
+  ### Configure ssh.
```
Comment (ming13) [inline README.md]: Formatting went south.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+  ```
+
+  * Specifying `PreferredAuthentications` speeds up connection phase.
+  * `ControlMaster` allows reusing existing ssh connections.
+  * `ControlPersist` specifies for how long ssh should keep connection open.
+
+* Copy and send public key to the person responsible for maintenance of remote build infrastructure.
+
+  ```
+  # On Mac OS you can copy content from commandline.
```
Comment (ming13) [inline README.md]: `macOS`

File: README.md [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+## Change compression level
+
+You can tune compression levels as you wish.
+
+For example, if your network is very slow, you might consider increasing compression level up to `9` to download/upload less data.
+If network is very fast, you might consider disable compression at all by passing `0` as a value (TODO). Default values are `1`.
+
+Configurable via properties in `local.properties`:
+
+```
```
Comment (ming13) [inline README.md]: `properties`

File: README.md [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+# Optional.
+remote_build.local_gzip_level=1
+
+# Optional.
+remote_build.remote_gzip_level=1
+```
+
+## Use best hardware available for remote machine
+
+Performant CPU, fast SSD, fast RAM (~8 GB per user), fast network.
```
Comment (ming13) [inline README.md]: Performant like what? CPU, cores, L3 cache?

File: README.md (hunk 3 above)
Comment (artem-zinnatullin) [inline README.md]: rili

File: README.md (hunk 5 above)
Comment (artem-zinnatullin) [inline README.md]: added

File: mainframer.sh [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+fi
+
+if [ -z "$REMOTE_GZIP_LEVEL" ]; then
+	REMOTE_GZIP_LEVEL=1
+fi
+
+BUILD_COMMAND="$@"
+
+if [ -z "$BUILD_COMMAND" ]; then
+	echo "Please pass build command as argument(s) to the script."
```
Comment (artem-zinnatullin) [inline mainframer.sh]: k

File: mainframer.sh (hunk 1 above)
Comment (artem-zinnatullin) [inline mainframer.sh]: Let's investigate `parallel`, I'd also like to gzip in parallel if possible. Created #6

File: mainframer.sh [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+cd ~ && \
+mkdir -p android-project-remote-build && \
+rm -rf android-project-remote-build/build/remotely_built_project.tar android-project-remote-build/*/src && \
+tar -xzf project_for_remote_build.tar -C android-project-remote-build && \
+cd android-project-remote-build && \
+$BUILD_COMMAND && \
+GZIP=-$REMOTE_GZIP_LEVEL tar -cz \
+--exclude='kotlin' \
+--exclude='tmp' \
+-f remotely_built_project.tar build/ */build"
```
Comment (artem-zinnatullin) [inline mainframer.sh]: PR welcome

File: mainframer.sh (hunk 1 above)
Comment (artem-zinnatullin) [inline mainframer.sh]: @dmitry-novikov will do in a separate PR
